libs/blocks/section-metadata/section-metadata.js

```diff
@@ -34,6 +36,39 @@ function handleTopHeight(section) {
   const headerHeight = document.querySelector('header').offsetHeight;
   section.style.top = `${headerHeight}px`;
 }
+
+function promoIntersectObserve(el, startPoint, options = {}) {
+  let abovePromoStart = false;
+  const io = new IntersectionObserver((entries, observer) => {
+    entries.forEach((entry) => {
+      if (el.classList.contains('close-sticky-section')) {
+        observer.unobserve(entry.target);
+        return;
+      }
+      const isPromoStart = startPoint && entry.target === startPoint;
+      abovePromoStart = isPromoStart
+        ? (entry.isIntersecting || entry.boundingClientRect.y > 0)
+        : abovePromoStart;
+      if (entry.isIntersecting || (isPromoStart && abovePromoStart)) el.classList.add('hide-sticky-section');
+      else if (!abovePromoStart) el.classList.remove('hide-sticky-section');
+    });
+  }, options);
+  return io;
+}
+
+function handleStickyPromobar(section) {
+  const main = document.querySelector('main');
+  section.classList.add('hide-sticky-section');
+  let startPoint = null;
+  if (main.children[0] !== section) {
+    startPoint = createTag('div', { class: 'section show-sticky-section' });
+    main.insertBefore(startPoint, section);
+  }
+  const io = promoIntersectObserve(section, startPoint);
+  if (startPoint) io.observe(startPoint);
+  io.observe(document.querySelector('footer'));
```

Comment:
If `startPoint === null` so there is no observation of `startPoint`, do we still need to observe the footer?

Also might be a bit clearer to name startPoint something like `stickySectionEl`

Reply:
@chrischrischris We have 2 use cases of sticky bottom promobar shared with us -
1. The promobar will be visible after a section is scrolled up and hides on footer
2. The promobar will be visible always and hides on footer
If our promobar is authored at the top then in condition 2 the startPoint will be null but we will still need to observe the footer to hide the promobar.

I have made the changes in the PR to change variable name from startPoint to stickySectionEl